src/components/organisms/Permission.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import { useWeb3 } from '../../providers/Web3'
+import rbacRequest from '../../utils/rbac'
+import Alert from '../atoms/Alert'
+import Loader from '../atoms/Loader'
+
+export default function Permission({
+  eventType,
+  children
+}: {
+  eventType: string
+  children: ReactElement
+}): ReactElement {
+  const [data, updateData] = useState<boolean>()
+  const { accountId } = useWeb3()
+  useEffect(() => {
+    const getData = async () => {
+      if (accountId !== undefined) {
+        const data = await rbacRequest(eventType, accountId)
+        updateData(data)
+      }
+    }
+    getData()
+  }, [eventType, accountId])
+  if (data === true) {
```

Comment:
we could have another check here to solve the isolation issue. If we check here for undefined `appConfig.rbacUrl` and immediately return the `children` our main use cases should not be affected anymore. Then in the above `useEffect` immediately `return` too when `appConfig.rbacUrl` is not present

Reply:
Good idea, I have implemented that here: https://github.com/oceanprotocol/market/pull/597/commits/22cacdb57747a700a84a5d3cc855b3fe7a50c1ca

Let me know if that was what you were thinking. As far as I can see that now makes the `if (url === undefined) ` check in rbac.ts redundant